tests/templates/test_integration.py

```diff
@@ -176,558 +138,331 @@
+             'varphi_2': qml.init.cvqnn_layers_varphi_normal(n_layers=3, n_wires=2),
+             'a': qml.init.cvqnn_layers_a_normal(n_layers=3, n_wires=2),
+             'phi_a': qml.init.cvqnn_layers_phi_a_normal(n_layers=3, n_wires=2),
+             'k': qml.init.cvqnn_layers_kappa_normal(n_layers=3, n_wires=2),
+             'wires': range(2)}),
+           (qml.templates.Interferometer,
+            {'phi': qml.init.interferometer_phi_uniform(n_wires=2),
+             'varphi': qml.init.interferometer_varphi_uniform(n_wires=2),
+             'theta': qml.init.interferometer_theta_uniform(n_wires=2),
+             'wires': range(2)}),
+           (qml.templates.Interferometer,
+            {'phi': qml.init.interferometer_phi_normal(n_wires=2),
+             'varphi': qml.init.interferometer_varphi_normal(n_wires=2),
+             'theta': qml.init.interferometer_theta_normal(n_wires=2),
+             'wires': range(2)}),
+           (qml.templates.Interferometer,
+            {'phi': qml.init.interferometer_phi_uniform(n_wires=3),
+             'varphi': qml.init.interferometer_varphi_uniform(n_wires=3),
+             'theta': qml.init.interferometer_theta_uniform(n_wires=3),
+             'wires': range(3)}),
+           (qml.templates.Interferometer,
+            {'phi': qml.init.interferometer_phi_normal(n_wires=3),
+             'varphi': qml.init.interferometer_varphi_normal(n_wires=3),
+             'theta': qml.init.interferometer_theta_normal(n_wires=3),
+             'wires': range(3)})
+           ]
+
+
+class TestIntegrationQnode:
+    """Tests the integration of templates into qnodes when differentiable arguments are passed as
+    primary or auxiliary arguments to the qnode, using different interfaces."""
+
+    @pytest.mark.parametrize("template, diffable, nondiffable", QUBIT_DIFFABLE_NONDIFFABLE)
     @pytest.mark.parametrize("interface, to_var", INTERFACES)
-    def test_integration_qubit_diffable(self, template1, diffable1, nondiffable1,
-                                        template2, diffable2, nondiffable2,
-                                        interface, to_var):
-        """Tests integration of qubit templates passing differentiable arguments as positional arguments to qnode."""
-        #TODO: rewrite test to avoid quadratic growth of test cases with the number of templates
+    def test_qubit_qnode_primary_args(self, template, diffable, nondiffable, interface, to_var):
+        """Tests integration of qubit templates with other operations, passing differentiable arguments
+        as primary arguments to qnode."""
 
         # Extract keys and items
-        keys_diffable1 = [*diffable1]
-        diffable1 = list(diffable1.values())
-        keys_diffable2 = [*diffable2]
-        diffable2 = list(diffable2.values())
-
-        # Combine diffable inputs to allow passing with *
-        diffable = diffable1 + diffable2
+        keys_diffable = [*diffable]
```

Comment:
Curious about the diff here: this seems to be addressing #562, if I understand correctly now only one template at a time is being tested, correct? Was there any reason (e.g. edge case) for testing each template pairwise before?

Reply:
Not at all, I just thought it was maximally diversifying. 

What I could do is use a hand-made template in the integration test, in addition to the gate.

Reply:
Implemented that!